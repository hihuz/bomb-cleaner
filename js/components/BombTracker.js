import React from 'react';
import { connect } from 'react-redux';

const BombTracker = (props) => (
  <div className="bomb-tracker">
    {props.bombsRemaining}
  </div>
);

const mapStateToProps = (state) => {
  return { bombsRemaining: state.grid.bombs - state.grid.flags };
};

export default connect(mapStateToProps)(BombTracker);
