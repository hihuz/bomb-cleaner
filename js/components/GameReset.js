import React from 'react';

const GameReset = props => (
  <div className="game-reset">
    <button className="reset-button" onClick={props.reset}>reset!</button>
  </div>
);

export default GameReset;
