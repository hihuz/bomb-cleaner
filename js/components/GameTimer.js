import React from 'react';

const GameTimer = props => (
  <div className="game-timer">
    {props.time}
  </div>
);

export default GameTimer;
