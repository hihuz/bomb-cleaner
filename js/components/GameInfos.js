import React from 'react';
import BombTracker from './BombTracker';
import GameReset from './GameReset';
import GameTimer from './GameTimer';

const GameInfos = ({ bombsRemaining, time, handleReset }) => (
  <div className="game-infos">
    <BombTracker bombsRemaining={bombsRemaining} />
    <GameReset reset={handleReset} />
    <GameTimer time={time} />
  </div>
);

export default GameInfos;
