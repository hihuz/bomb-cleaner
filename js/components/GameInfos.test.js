import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import GameInfos from './GameInfos';
import BombTracker from './BombTracker';
import GameReset from './GameReset';
import GameTimer from './GameTimer';

test('Snapshot', () => {
  const component = shallow(<GameInfos />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Should render a BombTracker component and pass it its bombsRemaining prop', () => {
  const bombs = 12;
  const component = shallow(<GameInfos bombsRemaining={bombs} />);
  const child = component.find(BombTracker);
  expect(child.length).toEqual(1);
  expect(child.props().bombsRemaining).toEqual(bombs);
});

test('Should render a GameReset component and pass it its handleReset prop', () => {
  const reset = 'boo';
  const component = shallow(<GameInfos handleReset={reset} />);
  const child = component.find(GameReset);
  expect(child.length).toEqual(1);
  expect(child.props().reset).toEqual(reset);
});

test('Should render a GameTimer component and pass it its time prop', () => {
  const time = 90;
  const component = shallow(<GameInfos time={time} />);
  const child = component.find(GameTimer);
  expect(child.length).toEqual(1);
  expect(child.props().time).toEqual(time);
});
