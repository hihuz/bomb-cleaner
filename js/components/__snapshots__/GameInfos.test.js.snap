exports[`test Snapshot 1`] = `
<div
  className="game-infos">
  <BombTracker />
  <GameReset />
  <GameTimer />
</div>
`;
